CMakeLists.txt

```diff
@@ -109,16 +109,23 @@ else()
 
   # No config file for this
   if(WITH_GFLAGS)
-    find_package(gflags REQUIRED)
+    option(GFLAGS_USE_TARGET_NAMESPACE "Use gflags import target with namespace." ON)
```

Comment:
```cmake
  set(GFLAGS_USE_TARGET_NAMESPACE ON CACHE BOOL "" FORCE)
```
would suffice

Reply:
Should we leave this to user in case they need it for whatever reason?
For example, if they want to deliver the built binary to system without new gflags (with namespacing available), `option` provides both our recommendation (ON) and freedom (set it to OFF without code change).

Reply:
i assume that you want to support gflags v2.2.1 and lower. as target namespace support was added in v2.2.2. see https://github.com/gflags/gflags/tree/v2.2.2. so instead of exposing a gflag option to user of rocksdb, i'd prefer having something like
```cmake
set(GFLAGS_USE_TARGET_NAMESPACE ON CACHE BOOL "" FORCE)
find_package(gflags REQUIRED)
if(NOT TARGET gflags::gflags)
  add_library (gflags:: gflags ALIAS gflags)
endif()
```

and do the same thing in #6791

Reply:
Note that CMake is stateful.
Injection to 3rd-party namespace can potentially mislead whatever `find_package()` call after `find_package(RocksDB)`, and should only be used if nothing else works.
In this case I think
```
if (...)
   list(APPEND ... gflags)
else()
   list(APPEND ... gflags::gflags)
endif()
```
should be the right way, which is equivalent to list(APPEND ... ${GFLAGS_TARGET})

But I agree that GFLAGS_USE_TARGET_NAMESPACE setting should also be written into config file.

Reply:
Regarding using `set(...)`, I'm confused by the `FORCE`.
`option(XXX)` doesn't make `XXX` a "RocksDB option", it's simply a cached variable and you can also have them show up in your build screen if created by one of you dependency config files. 

Image this use case:
```
option(GFLAGS_USE_TARGET_NAMESPACE "" OFF)
find_package(Foo) # Old lib Foo wants gflags without namespace.
find_package(RocksDB)
# Count-intuitive to say option(GFLAGS_USE_TARGET_NAMESPACE "" OFF) again.
find_package(Bar) # Old lib Bar also wants gflags without namespace.
```
Since RocksDB doesn't have any hard dependency on the namespaced gflags, it's better to leave that to user for compatibility.

Reply:
To summarize, this is what I'm trying to achieve:
1. Backward compatibility with old gflags (`${GFLAGS_TARGET}` for auto-fallback to `gflags` if `GFLAGS_USE_TARGET_NAMESPACE` is not recognized)
2. Forward compatibility (by not using `GFLAGS_USE_TARGET_NAMESPACE=OFF`) if namespace is required in the future since that's the trend of modern cmake.
3. Compatibility with other libs and allow user to decide if our choice is wrong.